sympy/core/basic.py

```diff
@@ -1071,7 +1074,7 @@ def sympify_new(new):
             else:
                 return sympify(new, strict=True)
 
-        sequence = [(sympify_old(s1), sympify_new(s2)) for s1, s2 in sequence]
+        sequence = [i.args if isinstance(i, Equality) else tuple(map(sympify_old, i)) for i in sequence]
```

Comment:
```suggestion
        sequence = [i.args if isinstance(i, Equality) else (sympify_old(i[0]), sympify_new(i[1])) for i in sequence]
```
This may have missed preserving `sympify_old, sympify_new` application exactly